src/Compilers/CSharp/Portable/Binder/Binder.cs

```diff
@@ -32,7 +32,7 @@ internal Binder(Binder next)
         internal Binder(CSharpCompilation compilation)
         {
             Debug.Assert(compilation != null);
-            this.Flags = BinderFlags.None;
+            this.Flags = compilation.Options.TopLevelBinderFlags;
```

Comment:
Thanks for adding a comment, as we discussed offline.  This constructor is only used by `BuckStopsHereBinder`.